OsmAnd/src/net/osmand/plus/GPXDatabase.java

```diff
@@ -201,17 +202,34 @@ public GpxDataItem(File file, int color) {
 
 		public GpxDataItem(File file, @NonNull GPXFile gpxFile) {
 			this.file = file;
+			readGpxParams(gpxFile);
+		}
+
+		private void readGpxParams(GPXFile gpxFile) {
 			color = gpxFile.getColor(0);
 			width = gpxFile.getWidth(null);
 			showArrows = gpxFile.isShowArrows();
 			showStartFinish = gpxFile.isShowStartFinish();
-			gradientScaleType = gpxFile.getGradientScaleType();
-			gradientSpeedColor = gpxFile.getGradientScaleColor(GradientScaleType.SPEED, 0);
-			gradientSlopeColor = gpxFile.getGradientScaleColor(GradientScaleType.SLOPE, 0);
-			gradientAltitudeColor = gpxFile.getGradientScaleColor(GradientScaleType.ALTITUDE, 0);
-			if (gpxFile.getSplitType() != null && gpxFile.getSplitInterval() != 0) {
-				splitType = gpxFile.getSplitType().getType();
-				splitInterval = gpxFile.getSplitInterval();
+			gradientSpeedColor = gpxFile.getGradientScaleColor(GradientScaleType.SPEED.getTypeName(), 0);
+			gradientSlopeColor = gpxFile.getGradientScaleColor(GradientScaleType.SLOPE.getTypeName(), 0);
+			gradientAltitudeColor = gpxFile.getGradientScaleColor(GradientScaleType.ALTITUDE.getTypeName(), 0);
+
+			if (!Algorithms.isEmpty(gpxFile.getSplitType()) && gpxFile.getSplitInterval() != 0) {
+				for (GpxSplitType gpxSplitType : GpxSplitType.values()) {
+					if (gpxSplitType.name().equalsIgnoreCase(gpxFile.getSplitType())) {
```

Comment:
Methods for getting Type by name must be in enums!
GradientScaleType
GpxSplitType

Reply:
Methods added to enums.